solutions/swb-app/src/dataSets/DataSetFileUploadRequestParser.ts

```diff
@@ -2,12 +2,12 @@
  *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
  *  SPDX-License-Identifier: Apache-2.0
  */
-import { z } from 'zod';
+import { z } from '@aws/workbench-core-base';
 
 // eslint-disable-next-line @rushstack/typedef-var
 export const DataSetFileUploadRequestParser = z.object({
-  dataSetId: z.string(),
-  filenames: z.union([z.string(), z.array(z.string())])
+  dataSetId: z.string().datasetId().required(),
+  filenames: z.union([z.string().swbName().required(), z.array(z.string().swbName().required()).min(1)])
```

Comment:
nit: require() is checking min(1). you don't need both

Reply:
Here the `required()` is on `z.string()` and `min(1)` is on `z.array()` as `z.array()` is not compatible with `required()`. I was not certain if they mean the same thing in this scenario. Do you know for certain?

Reply:
good callout!! I didn't realized min(1) is on array.